### ISSUE TITLE
4.1 build without Firebase

### ISSUE THREAD

**Issue by licaon-kter** [BelledonneCommunications/linphone-android#482]
Trying to prepare for [4.1 on F-Droid](https://gitlab.com/fdroid/fdroiddata/merge_requests/4524)

I'm returning `false` [here](https://github.com/BelledonneCommunications/linphone-android/blob/feature/release-4.1/app/build.gradle#L9) and removing the lib [here](https://github.com/BelledonneCommunications/linphone-android/blob/feature/release-4.1/app/build.gradle#L139)

I'm removing [this file](https://github.com/BelledonneCommunications/linphone-android/tree/feature/release-4.1/app/src/main/java/org/linphone/firebase)

I'm setting `false` [here](https://github.com/BelledonneCommunications/linphone-android/blob/feature/release-4.1/app/src/main/res/values/non_localizable_custom.xml#L67) and `none` [here](https://github.com/BelledonneCommunications/linphone-android/blob/feature/release-4.1/app/src/main/res/values/non_localizable_custom.xml#L69)

So I'm thinking this is it, Firebase is gone, Push is set to disable in app (not that it matters since it won't work, right?)

But:
* build fails since [4 Firebase related imports](https://github.com/BelledonneCommunications/linphone-android/blob/feature/release-4.1/app/src/main/java/org/linphone/LinphoneManager.java#L59) aren't available
* build fails since [Firebase related code](https://github.com/BelledonneCommunications/linphone-android/blob/feature/release-4.1/app/src/main/java/org/linphone/LinphoneManager.java#L731) isn't available
* in app Push checkbox is checked anyway, for some reason

So my question, can this be fixed in the app code?

I can just [remove the imports and gut the `try{}` to `return;`](https://gitlab.com/fdroid/fdroiddata/merge_requests/4524/diffs#247e3f6ae19746070be46d0ec342bd42a7b76169_0_1) but this will need to be adapted on each new version, I was even hoping we could set the app to autoupdate on F-Droid.

Thanks

**Comment by Viish** [comment]
Hi,
I'll try to have a simpler way to disable firebase.
I'll keep you updated.
Thanks for the report.

**Comment by Viish** [comment]
By the way, why do you remove the lib ?
It you keep it you won't have the import problem anymore and you could keep the FirebaseService.
And you aren't required to set none for push_type if it is disabled.

**Comment by licaon-kter** [comment]
> By the way, why do you remove the lib ?

It's not open-source.

F-Droid either builds from source or pulls artefacts of open-source libs from approved maven repos.

**Comment by Viish** [comment]
commit feb6312c93769fcf55938e711dc47da4e0753f75 (HEAD -> feature/release-4.1, origin/feature/release-4.1)
Author: Sylvain Berfini <sylvain.berfini@belledonne-communications.com>
Date:   Fri Feb 22 11:04:02 2019 +0100
    If google-services.json file is not found, completely disable firebase dependency

**Comment by gdt** [comment]
This was closed.  Does that mean that there is a release with the problem fixed?

**Comment by licaon-kter** [comment]
@gdt Nope, I do all these and one more thing when building and the push checkbox is still there checked on install.

**Comment by Viish** [comment]
Yes the push checkbox isn't currently linked to the presence of the google-services.json file.

**Comment by licaon-kter** [comment]
> Do you know an API that would allow to know if the device supports FCM ?

At build time yes, that file missing should trigger a push checkbox disable/hide.

Google Services missing or disabled are a sign that you can't have push too.

**Comment by Viish** [comment]
File missing at build time ok, I will make the change.
Problem is how to detect an app that has been built with firebase support but that is running on a device that doesn't supports it.

**Comment by licaon-kter** [comment]
@Viish 
> Google Services missing or disabled are a sign that you can't have push too.

^^^^ you can check this.

**Comment by Viish** [comment]
The question was how...
Don't mind, I found the answer on stackoverflow.

**Comment by licaon-kter** [comment]
@Viish Yeah, better you than me :)

**Comment by licaon-kter** [comment]
@Viish I'll compile later (it takes a while with the SDK), but these: https://github.com/BelledonneCommunications/linphone-android/commit/767254b64a83d5bdb9d802ebeb6dda91c4e9b37e#diff-bd3bf79f1c3774f17d234a5ca2906e97 won't fail on F-Droid since it doesn't load the "gms"-libs?

**Comment by Viish** [comment]
I don't think so, the purpose of this API is to know whether it is available or not. If it fails because it is not available then what's the point of having created them ?

**Comment by licaon-kter** [comment]
The API is not the point, I was just saying that **now** the app can't be build using only free open-source libraries.

/PS: GMS is closed sourced

**Comment by Viish** [comment]
Why do you say that ? You can still build it without the push notifs framework.

**Comment by licaon-kter** [comment]
As the OP of the issue, building for F-Droid, only with free open-source libs:
```
/home/android/fdroiddata-0402/build/org.linphone/app/src/main/java/org/linphone/utils/DeviceUtils.java:37: error: package com.google.android.gms.common does not exist
import com.google.android.gms.common.ConnectionResult;
                                    ^
/home/android/fdroiddata-0402/build/org.linphone/app/src/main/java/org/linphone/utils/DeviceUtils.java:38: error: package com.google.android.gms.common does not exist
import com.google.android.gms.common.GoogleApiAvailability;
```
...will fail **now**, since Google services are not... open-source...

**Comment by Viish** [comment]
OK thanks I didn't understood.
Should be fixed by 1cfdc5c2ffa2e790b2a1fb0646ffa7b96c42e396
